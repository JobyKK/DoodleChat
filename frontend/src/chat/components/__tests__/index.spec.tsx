import React from 'react';
import { Provider } from 'react-redux';
import {Chat} from './../index.tsx';
import configureStore from 'redux-mock-store';

let props = {};
const mockStore = configureStore();

beforeEach(() => {
  props = {
    user: {},
    messages: [],
    sendMessage: jest.fn(),
    subscribeOnMessages: jest.fn(),
    getAllMessages: jest.fn(),
    classes: {}
  };
});

describe('Chat component', () => {
  [{}, undefined].forEach(user => {
    it(`check the login menu, where user = ${user}`, () => {
      props.user = user;

      const wrapper = renderer.create(
        <Provider store={mockStore({})}><Chat {...props} /></Provider>
      ).toJSON();
  
      expect(wrapper).toMatchSnapshot();
    });
  })
});
