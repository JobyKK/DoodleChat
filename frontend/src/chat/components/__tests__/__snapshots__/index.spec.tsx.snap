// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component check the login menu, where user = [object Object] 1`] = `
<div>
  <div />
  <div
    className="InputBlock-block-1"
  >
    <input
      className="InputBlock-input-3"
      name="text"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Message"
      type="text"
      value=""
    />
    <button
      className="InputBlock-button-2"
      onClick={[Function]}
    >
      Send
    </button>
  </div>
</div>
`;

exports[`Chat component check the login menu, where user = undefined 1`] = `
<div>
  <div />
  <div
    className="InputBlock-block-1"
  >
    <input
      className="InputBlock-input-3"
      name="text"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <button
      className="InputBlock-button-2"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
